storage/test/native_rocksdb_client_test.cpp

```diff
@@ -0,0 +1,621 @@
+  ASSERT_THROW(db->put(cf, key, value), RocksDBException);
+  ASSERT_THROW(db->del(cf, key), RocksDBException);
+}
+
+TEST_F(native_rocksdb_test, put_in_default_family) {
+  db->put(key, value);
+
+  // Ensure key is in the default family via the non-family interface.
+  {
+    const auto dbValue = db->get(key);
+    ASSERT_TRUE(dbValue.has_value());
+    ASSERT_EQ(*dbValue, value);
+  }
+
+  // Ensure key is in the default family via the family interface.
+  {
+    const auto dbValue = db->get(db->defaultColumnFamily(), key);
+    ASSERT_TRUE(dbValue.has_value());
+    ASSERT_EQ(*dbValue, value);
+  }
+}
+
+TEST_F(native_rocksdb_test, put_with_different_key_and_value_types) {
+  db->put("key"sv, "value"s);
+  const auto dbValue = db->get("key"s);
+  ASSERT_TRUE(dbValue.has_value());
+  ASSERT_EQ(*dbValue, "value");
+}
+
+TEST_F(native_rocksdb_test, del_from_default_family) {
+  db->put(key, value);
+  db->del(key);
+  ASSERT_FALSE(db->get(key).has_value());
+}
+
+TEST_F(native_rocksdb_test, put_in_created_family) {
+  const auto cf = "cf"s;
+  db->createColumnFamily(cf);
+  db->put(cf, key, value);
+  const auto dbValue = db->get(cf, key);
+  ASSERT_TRUE(dbValue.has_value());
+  ASSERT_EQ(*dbValue, value);
+  // Ensure the key is not the default family.
+  ASSERT_FALSE(db->get(key).has_value());
+}
+
+TEST_F(native_rocksdb_test, del_from_created_family) {
+  const auto cf = "cf"s;
+  db->createColumnFamily(cf);
+  // Put in both the default and the created one.
+  db->put(key, value);
+  db->put(cf, key, value);
+  db->del(cf, key);
+  const auto defaultValue = db->get(key);
+  ASSERT_TRUE(defaultValue.has_value());
+  ASSERT_EQ(*defaultValue, value);
+  ASSERT_FALSE(db->get(cf, key).has_value());
+}
+
+TEST_F(native_rocksdb_test, del_non_existent_key_is_not_an_error) { ASSERT_NO_THROW(db->del(key)); }
+
+TEST_F(native_rocksdb_test, drop_family_and_get) {
+  const auto cf = "cf"s;
+  db->createColumnFamily(cf);
+  db->put(cf, key, value);
+  db->dropColumnFamily(cf);
+  ASSERT_THROW(db->get(cf, key), RocksDBException);
+}
+
+TEST_F(native_rocksdb_test, drop_family_then_create_and_ensure_empty) {
+  const auto cf = "cf"s;
+  db->createColumnFamily(cf);
+  db->put(cf, key, value);
+  db->dropColumnFamily(cf);
+  db->createColumnFamily(cf);
+  ASSERT_FALSE(db->get(cf, key).has_value());
+}
+
+TEST_F(native_rocksdb_test, put_in_batch_in_default_family) {
+  auto batch = db->getBatch();
+  batch.put(key1, value1);
+  batch.put(key2, value2);
+  db->write(std::move(batch));
+  const auto dbValue1 = db->get(key1);
+  const auto dbValue2 = db->get(key2);
+  ASSERT_TRUE(dbValue1.has_value());
+  ASSERT_EQ(*dbValue1, value1);
+  ASSERT_TRUE(dbValue2.has_value());
+  ASSERT_EQ(*dbValue2, value2);
+}
+
+TEST_F(native_rocksdb_test, put_in_batch_in_2_families) {
+  const auto cf1 = "cf1"s;
+  const auto cf2 = "cf2"s;
+  db->createColumnFamily(cf1);
+  db->createColumnFamily(cf2);
+  auto batch = db->getBatch();
+  db->put(cf1, key1, value1);
```

Comment:
Don't these need to be `batch.put` calls?

Reply:
What a great catch, thanks, fixing!